collections/src/main/scala/strawman/collection/StrictOptimizedSeqOps.scala

```diff
@@ -16,11 +16,7 @@ trait StrictOptimizedSeqOps [+A, +CC[_], +C]
     val seen = mutable.HashSet.empty[B]
 
     for (x <- this) {
-      val y = f(x)
-      if (!seen.contains(y)) {
-        seen += y
-        builder += x
-      }
+      if (seen.add(f(x))) builder += x
```

Comment:
Is there any way to return the same collection in case nothing was changed or if there is one 0/1 element?
In fact few other functions here could also benefit from this optimization.

Reply:
`coll` is the new equivalent of the former `repr`.

Reply:
Thanks!